src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,101 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
+// eslint-disable-next-line max-len
+import { addComment, getPostComments, removeComments } from '../../api/comments';
+import { getPost } from '../../api/posts';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+export const PostDetails = ({ postId }) => {
+  const [post, setPost] = useState(null);
+  const [comments, setComments] = useState([]);
+  const [commentsVisible, setCommentsVisible] = useState(true);
+
+  useEffect(() => {
+    if (!postId) {
+      setPost(null);
+
+      return;
+    }
+
+    loadPostAndComments(postId);
+  }, [postId]);
+
+  const loadPostAndComments = async(id) => {
+    const [postServer, commentsServer] = await Promise.all(
+      [getPost(id), getPostComments(id)],
+    );
+
+    setPost(postServer);
+    setComments(commentsServer);
+  };
+
+  const removeHandler = async(commentId) => {
+    await removeComments(commentId);
+    setComments(com => com.filter(({ id }) => commentId !== id));
+  };
+
+  const addNewComment = (newComment) => {
+    addComment(newComment)
+      .then(addedCom => setComments(com => [...com, addedCom]));
+  };
+
+  return (
+    <div className="PostDetails">
+
+      {post ? (
+        <>
+          <h2>Post details:</h2>
+          <section className="PostDetails__post">
+            <p>{post.body}</p>
+          </section>
+
+          <section className="PostDetails__comments">
+            <button
+              type="button"
+              onClick={() => setCommentsVisible(value => !value)}
+              className="button"
+            >
+              {commentsVisible ? 'Hide' : 'Show'}
+              {' '}
+              {comments.length}
+              {' '}
+              comments
```

Comment:
```suggestion
              {commentsVisible
                ? `Hide ${comments.length} comments`
                : `Show  ${comments.length} comments`
              }
```
Иногда стоит продублировать текст, чтобы повысить наглядность 
А если ты захочешь написать вообще разные фразы то это станет проще сделать